src/natural_language_geocoding/models.py

```diff
@@ -13,37 +13,38 @@
 from shapely.geometry.base import BaseGeometry
 
 from natural_language_geocoding.natural_earth import coastline_of
-from natural_language_geocoding.nominatim import nominatim_search
+from natural_language_geocoding.place_lookup import PlaceLookup
 
 
 class SpatialNodeType(BaseModel, ABC):
     model_config = ConfigDict(strict=True, extra="forbid", frozen=True)
 
     @abstractmethod
-    def to_geometry(self) -> BaseGeometry | None: ...
+    def to_geometry(self, place_lookup: PlaceLookup) -> BaseGeometry | None: ...
 
 
-class NamedEntity(SpatialNodeType):
+class NamedPlace(SpatialNodeType):
     """Represents the name of a place somewhere in the world."""
 
-    node_type: Literal["NamedEntity"] = "NamedEntity"
+    node_type: Literal["NamedPlace"] = "NamedPlace"
     name: str
 
     subportion: Literal["western half", "northern half", "southern half", "eastern half"] | None = (
         Field(
             default=None,
             description=singleline(
                 """
-                An optional field to indicate that a subportion of the NamedEntity is referenced
+                An optional field to indicate that a subportion of the NamedPlace is referenced
                 suchas "Western Brazil" would refer to the west half of Brazil. Note this is NOT
-                used in cases where a cardinal direction is part of the place name like "South Africa"
+                used in cases where a cardinal direction is part of the place name like
+                "South Africa"
             """
             ),
```

Comment:
Typo: "suchas". Suggestion: '... the NamedPlace is referenced. For example, "Western Brazil" would refer to ...'.

This is subjective but the way I like to format long descriptions is by breaking them into multiple strings and taking advantage of the fact that Python automatically concatenates adjacent strings:
```py
        Field(
            default=None,
            description="An optional field to indicate that a subportion of the NamedPlace is "
            'referenced. For example, "Western Brazil" would refer to the west half of Brazil. '
            "Note this is NOT used in cases where a cardinal direction is part of the place "
            'name like "South Africa".',
        )
```

Reply:
I tried the quotes around everything way but it gets finicky when you want to edit it. If you start modifying it in the middle then rewrapping and fixing all the quotes after that point is required. With the """ approach you only have to adjust where the newlines are.